nix/ctl-server-nixos-module.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
```

Comment:
Can we have some CI check to ensure this code will be up to date?

Reply:
What do you mean?

Reply:
I mean, right now it's not actually executed in CI, right?

Reply:
It is, this PR also adds a new `nixosConfiguration` called `test` which is a minimal VM that makes use of the new NixOS modules. Moreover this `nixosConfiguration` is also added to `hydraJobs`.